server/controllers/usercontroller.js

```diff
@@ -48,18 +49,94 @@ class User {
   }
 
   static async accountVerification(req, res) {
-      const { token } = req.params;
-      const getInfo = jwt.verify(token, process.env.JWT_KEY);
+    const { token } = req.params;
+    const getInfo = jwt.verify(token, process.env.JWT_KEY);
+
+    await userService.updateAtt(
+      { isVerified: true },
+      { email: getInfo.email },
+    );
+    
+    const message = 'Account was successfully verified.';
+
+    util.setSuccess(200, message);
+    return util.send(res);
+  }
+
+  static async Oauth(req,res){
+    let Action;
+    let status;
+    let userGot;
+    let google = null;
+    let facebook = null;
+    const column = `${req.user.provider}`;
 
-      await userService.updateAtt(
-        { isVerified: true },
-        { email: getInfo.email },
-      );
-      
-      const message = 'Account was successfully verified.';
+    if(column.includes('google')){
+      google = req.user.id;
+    }else{
+      facebook = req.user.id;
+    }
+
+    const googleSearch = await userService.findByProp({
+      googleId : `${req.user.id}`,
+    })
+
+    const facebookSearch = await userService.findByProp({
+      facebookId : `${req.user.id}`,
+    })
+    
+    
+    const exist = await userService.findByProp({
+        email: `${req.user.emails[0].value}`,
+    })
 
-      util.setSuccess(200, message);
-      return util.send(res);
+    if (googleSearch[0]||facebookSearch[0]) {
+      if(!googleSearch[0]){
+        facebookSearch[0].dataValues
+      }else{
+        googleSearch[0].dataValues
+    }
+    Action = 'Log In';
+    status = 200;
+      }else if(exist[0]){
```

Comment:
All checks should be in the middleware

Reply:
That note a response, its just setting values that will be response according to the action made